glide-core/src/client/value_conversion.rs

```diff
@@ -640,6 +731,230 @@ mod tests {
+        // test convert nil is OK
+        assert_eq!(
+            convert_to_expected_type(Value::Nil, Some(ExpectedReturnType::XreadReturnType)),
+            Ok(Value::Nil)
+        );
+
+        // in RESP2, we get an array of arrays value like this:
+        // 1) 1) "key1"
+        //    2) 1) 1) "streamid-1"
+        //          2) 1) "f1"
+        //             2) "v1"
+        //             3) "f2"
+        //             4) "v2"
+        //       2) 1) "streamid-2" ...
+        // 2) 1) "key2"...
+
+        // test convert array to xread_map is OK
+        let array_of_arrays = vec![
+            Value::Array(vec![
+                Value::BulkString(b"key1".to_vec()),
+                Value::Array(vec![Value::Array(vec![
+                    Value::BulkString(b"streamid-1".to_vec()),
+                    Value::Array(vec![
+                        Value::BulkString(b"field1".to_vec()),
+                        Value::BulkString(b"value1".to_vec()),
+                    ]),
+                ])]),
+            ]),
+            Value::Array(vec![
+                Value::BulkString(b"key2".to_vec()),
+                Value::Array(vec![
+                    Value::Array(vec![
+                        Value::BulkString(b"streamid-2".to_vec()),
+                        Value::Array(vec![
+                            Value::BulkString(b"field21".to_vec()),
+                            Value::BulkString(b"value21".to_vec()),
+                            Value::BulkString(b"field22".to_vec()),
+                            Value::BulkString(b"value22".to_vec()),
+                        ]),
+                    ]),
+                    Value::Array(vec![
+                        Value::BulkString(b"streamid-3".to_vec()),
+                        Value::Array(vec![
+                            Value::BulkString(b"field3".to_vec()),
+                            Value::BulkString(b"value3".to_vec()),
+                        ]),
+                    ]),
+                ]),
+            ]),
+        ];
+
+        let converted_map = convert_to_expected_type(
+            Value::Array(array_of_arrays),
+            Some(ExpectedReturnType::XreadReturnType),
+        )
+        .unwrap();
+
+        let converted_map = if let Value::Map(map) = converted_map {
+            map
+        } else {
+            panic!("Expected a Map, but got {:?}", converted_map);
+        };
+        // expect 2 keys
+        assert_eq!(converted_map.len(), 2);
+
+        let (key, value) = &converted_map[0];
+        assert_eq!(*key, Value::BulkString(b"key1".to_vec()));
+        assert_eq!(
+            *value,
+            Value::Map(vec![(
+                Value::BulkString(b"streamid-1".to_vec()),
+                Value::Array(vec![
+                    Value::BulkString(b"field1".to_vec()),
+                    Value::BulkString(b"value1".to_vec()),
+                ]),
+            ),]),
+        );
+
+        // convert to a map value like this:
```

Comment:
nit: Since we do the conversion above, can we move this comment farther up (up 1 or 2 code blocks)? Seems like the comment belongs there rather than between the assertion of the first and second maps